### ISSUE TITLE
Move relevant developer's guide info from confluence to doxygen/readme/API Guide

### ISSUE THREAD

**Issue by jenhicks** [opensim-org/opensim-core#2254]
https://simtk-confluence.stanford.edu:8443/display/OpenSim/Introduction+to+the+OpenSim+API
For this page, remove/make private the page on "The OpenSim API" after making sure any relevant info is included in the API guide or other pages.

https://simtk-confluence.stanford.edu:8443/display/OpenSim/Developer%27s+Guide
Move any relevant background info in this section to the readme or API guide or other parts of confluence. 
Make the documentation for specific examples hidden or move it to md pages with the API code examples on github.
Update this page: remove the links to the confluence pages for each example and point to where to find the code instead.

**Comment by jenhicks** [comment]
Here are some more details about what I'm thinking @nickbianco @jimmyDunne @chrisdembia @aseth1 

1) Move contents of
https://simtk-confluence.stanford.edu:8443/display/OpenSim/Technical+Background
and
https://simtk-confluence.stanford.edu:8443/display/OpenSim/Guide+to+Using+Doxygen

to this page:

https://simtk-confluence.stanford.edu:8443/display/OpenSim/Getting+Started+as+a+Developer
(first section of developer's guide)

2) Keep this in the developer's guide on confluence?
https://simtk-confluence.stanford.edu:8443/display/OpenSim/Prerequisites

3) Remove this page (it replicates what's on github)
https://simtk-confluence.stanford.edu:8443/display/OpenSim/Step-by-Step+Example

4) Eliminate/hide this whole section after offloading the relevant content
https://simtk-confluence.stanford.edu:8443/display/OpenSim/Introduction+to+the+OpenSim+API

5) Hide and/or move to an md file in the repo: This page and all subsequent sections of developer's guide:
https://simtk-confluence.stanford.edu:8443/display/OpenSim/Performing+a+Simulation

6) Include a  list of available API examples and where to find them in the confluence developer's guide

7) Include links to the readme and API guide on github in the confluence developer's guide

**Comment by nickbianco** [comment]
> 2. Keep this in the developer's guide on confluence?
https://simtk-confluence.stanford.edu:8443/display/OpenSim/Prerequisites

We could probably hide/remove this page, since instructions for setting up development environments is already on the GitHub README. We'd could add a link to the README to direct users to pre-built binaries if they'd prefer that over building from source.

We could also move the sections "What is an OpenSim plug-in?" and "What is an OpenSim "main" program" from https://simtk-confluence.stanford.edu:8443/display/OpenSim/The+OpenSim+API to the Technical Background page. The rest is already in the API guide on GitHub.

**Comment by jenhicks** [comment]
Sounds good @nickbianco . Let's see if others have additional input, then we could work on implementing tomorrow.

**Comment by chrisdembia** [comment]
(1) Technical Background and Guide to Using Doxygen might be useful for scripting users too, so I'm not sure they should be moved to the Developer's Guide.

(2) I agree with @nickbianco . It may be useful to still have a place where we have links to documentation for CMake, Visual Studio, in case people want to get some background on what these tools are.

(3) @jenhicks I think you're comparing this to the README on GitHub. There's a difference: the README describes how to build OpenSim itself, while this page describes how to build a C++ project that depends on OpenSim (something students in 485 have to do, even if they don't build OpenSim itself). Perhaps this should be moved to the Doxygen or the API Guide.

I agree with all other suggestions.

**Comment by jenhicks** [comment]
There are two items to address as noted in my comment above.

**Comment by chrisdembia** [comment]
I updated the API Examples page by updating the list of examples (and saying where to find them) and removing the CMake and Visual Studio instructions.

I renamed two pages:
- Step-by-Step Example -> How to Build a C++ Example
- API Examples -> C++ API Examples

The code file for Performing a Simulation (ExampleMain) is pretty much empty. It's an "interactive" example where the user follows the now-hidden Confluence page to fill in the code file. I suggest we  just update the code file in the repo to contain the completed code. For now, I updated the Confluence page to refer users to the OutputReference, which contains the complete code.
